src/coreclr/vm/prestub.cpp

```diff
@@ -2369,12 +2376,27 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT, CallerGCMode callerGCMo
     MemoryBarrier();
 #endif
 
+#if defined(FEATURE_PORTABLE_ENTRYPOINTS)
+    if (pStub != NULL)
+    {
+        pCode = pStub->GetEntryPoint();
+        pStub->DecRef();
+        delete pStub;
+        pStub = NULL;
+
+        void* ilStubInterpData = PortableEntryPoint::GetInterpreterData(pCode);
+        _ASSERTE(ilStubInterpData != NULL);
+        SetInterpreterCode((InterpByteCodeStart*)ilStubInterpData);
+    }
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
+
     if (pCode != (PCODE)NULL)
     {
         _ASSERTE(!MayHaveEntryPointSlotsToBackpatch()); // This path doesn't lock the MethodDescBackpatchTracker as it should only
                                                         // happen for jump-stampable or non-versionable methods
         SetCodeEntryPoint(pCode);
     }
+#ifndef FEATURE_PORTABLE_ENTRYPOINTS
     else
     {
         if (!GetOrCreatePrecode()->SetTargetInterlocked(pStub->GetEntryPoint()))
```

Comment:
```suggestion
else
{
#ifdef FEATURE_PORTABLE_ENTRYPOINTS
        pCode = pStub->GetEntryPoint();
        pStub->DecRef();

        void* ilStubInterpData = PortableEntryPoint::GetInterpreterData(pCode);
        _ASSERTE(ilStubInterpData != NULL);
        SetInterpreterCode((InterpByteCodeStart*)ilStubInterpData);
#else
        if (!GetOrCreatePrecode()->SetTargetInterlocked(pStub->GetEntryPoint()))
```
Move this here to have just have one ifdef?

Reply:
I guess we can do this if we add call to `SetCodeEntryPoint(pCode);`